torch/linalg/__init__.py

```diff
@@ -481,33 +514,44 @@
 """)
 
 eigvalsh = _add_docstr(_linalg.linalg_eigvalsh, r"""
-linalg.eigvalsh(input, UPLO='L', *, out=None) -> Tensor
+linalg.eigvalsh(A, UPLO='L', *, out=None) -> Tensor
 
-Computes the eigenvalues of a complex Hermitian or real symmetric matrix. Supports batched inputs.
+Computes the eigenvalues of a complex Hermitian or real symmetric matrix.
 
-The eigenvalues are returned in ascending order.
+Letting :math:`\mathbb{K}` be :math:`\mathbb{R}` or :math:`\mathbb{C}`,
+the eigenvalues of a square matrix :math:`A \in \mathbb{K}^{n \times n}` are defined
+as the roots (counted with multiplicity) of the polynomial of degree `n`
+
+.. math::
+
+    p(\lambda) = \operatorname{det}(A - \lambda \mathrm{I}_n)
 
-:attr:`input` is assumed to be Hermitian (resp. symmetric), but this is not checked internally.
-When :attr:`UPLO` is `'L'` (default), only the lower triangular part of each matrix is used in the computation.
-When :attr:`UPLO` is `'U'`, only the upper triangular part of each matrix is used.
+where :math:`\mathrm{I}_n` is the `n`-dimensional identity matrix.
+The eigenvalues of a real symmetric or complex Hermitian matrix are always real.
 
-Supports :attr:`input` of float, double, cfloat and cdouble dtypes.
-The output tensor will always be real-valued, even when :attr:`input` is complex.
+Supports inputs of float, double, cfloat and cdouble dtypes.
+Also supports batched inputs, and, if the input is batched, the output is batched with the same dimensions.
+
+:attr:`A` is assumed to be Hermitian (resp. symmetric), but this is not checked internally, instead:
+
+- If :attr:`UPLO` `= 'L'` (default), only the lower triangular part of the matrix is used in the computation.
+- If :attr:`UPLO` `= 'U'`, only the upper triangular part of the matrix is used.
+
+The returned tensor will always be real-valued, even when :attr:`A` is complex.
```

Comment:
I would try to combine these sentences and connect "tensor" with "eigenvalues":

"The returned eigenvalues will always be real-valued, even when :attr:\`A\` is complex, and sorted in ascending order."

Reply:
These sentences have been moved to the `Returns` section.